theforeman.org/pipelines/test/smart-proxy.groovy

```diff
@@ -67,7 +67,12 @@ pipeline {
                 axes {
                     axis {
                         name 'ruby'
-                        values '2.7'
+                        values '2.7', '3.0', '3.1'
+                    }
+                }
+                when {
+                    expression {
+                        ruby_versions.contains(env.ruby)
```

Comment:
I get the impression you're copying this from `foreman.groovy`, but there it's defined in another file: https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/test/foreman/vars.groovy

Then it's set at the start of the file:
https://github.com/theforeman/jenkins-jobs/blob/39e5d043a8ab292206210577cf3d0fb572ed387a/theforeman.org/pipelines/test/foreman/unit.groovy#L2

In this case you don't, so it will fail. I think it's safe to leave out the `when` expression.